server/middleware/authentication.js

```diff
@@ -371,3 +372,32 @@ export function mustBeLoggedIn(req, res, next) {
     }
   });
 }
+
+/**
+ * Checks if the user has two-factor authentication enabled on their account,
+ * and if they do, we challenge them to auth with it during the login flow
+ */
+export const twoFactorAuthCheck = (req, res, next) => {
```

Comment:
Introducing this separate function from `updateToken` is creating complexities that could be avoided (such as `res.locals` usage).

I would suggest to move everything in `updateToken` and if necessary extract some parts of it in an helper function.